addons/slack/uninstall.js

```diff
@@ -0,0 +1,36 @@
+/*
+This is the uninstallation logic of the Lifecycle Manager. 
+*/
+
+const Sdk = require('@fusebit/add-on-sdk');
+const Superagent = require('superagent');
+
+module.exports = async (ctx) => {
+  // Destroy the Add-On Handler
+  let functionCtx = await Sdk.deleteFunction(ctx);
```

Comment:
What is functionCtx, and does it get returned if the function we are trying to delete no longer exists?

Consider this: 
1. uninstall is called, delete function succeeds, but deregistration fails for some reason
2. Folks try to call uninstall again to arrive at an eventually consistent place - if the functionCtx is not established properly despite the absence of the add-on function, the deregistration will never succeed.

Reply:
It's the function metadata obtained via a GET to the function definition. If the function is already deleted, it will fail. As currently implemented, if deregistration fails, then the uninstall will error and the user will see it (and hopefully report it to us). The function and storage will already be deleted, so there won't be any visible additional artifacts for them to clean up and no entry point to retry uninstall.

Overall if the function config is gone we don't have the URI and token to anyway, so there's nothing we can do. If you want to optimize for cleaning up, we should move this before the function and storage delete, in which case it will be possible to rerun this.

Reply:
I think what would be more robust and reentrant is: 
1. Get function config
2. Call it to deregister (success if already deregistered)
3. Delete storage artifacts
4. Delete function

Reply:
Fixed